### PR TITLE
Added testcase for WebCache with checkpoint and restore

### PR DESCRIPTION
**Pull request by KeenanMcCarthy**

### CODE REVIEW

**Comment by KeenanMcCarthy** [comment]
#build
#libby

File: dev/io.openliberty.checkpoint_fat/publish/servers/webCacheServer/bootstrap.properties [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+bootstrap.include=../testports.properties
+websphere.java.security.exempt=true
+com.ibm.ws.logging.trace.specification=*=info:checkpoint=all
+io.openliberty.checkpoint.dump.threads=true
+io.openliberty.checkpoint.allowed.features=webCache-1.0
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/publish/servers/webCacheServer/bootstrap.properties:17]: Must enable webCache-1.0 by adding the header `WLP-InstantOn-Enabled: true` to https://github.com/OpenLiberty/open-liberty/blob/fcb5a7c86f4329639936c3bda3cc18fd58fa8b51/dev/com.ibm.websphere.appserver.features/visibility/public/webCache-1.0/com.ibm.websphere.appserver.webCache-1.0.feature

File: dev/io.openliberty.checkpoint_fat/bnd.bnd [hunk 2]
```diff
@@ -46,6 +46,7 @@ src: \
 	test-applications/virtualhost/src,\
 	test-applications/DefaultPrincipalMappingApp/src,\
 	test-applications/concurrentApp/src
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/bnd.bnd]: Need a `,\` after `test-applications/concurrentApp/src` to pickup the new webCacheApp source path

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.HttpUtils;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class WebCacheTest {
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java:44]: This needs to be added to the `FATSuite` class to run

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        long endTime = System.currentTimeMillis();
+        return (endTime - startTime) < CACHE_MISS_THRESHHOLD;
+    }
+
+    @Test
+    public void testWebCacheServlet() throws Exception {
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS);
+        server.startServer();
+        server.stopServer();
```
Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java]: calling `stopServer()` followed by `checkpointRestore()` should be needed with the simple setCheckpoint call above which will auto restore on `startServer`

File: dev/io.openliberty.checkpoint_fat/bnd.bnd (hunk 2 above)
Comment (KeenanMcCarthy) [reply dev/io.openliberty.checkpoint_fat/bnd.bnd]: Resolved

File: dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java (hunk 3 above)
Comment (KeenanMcCarthy) [reply dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/WebCacheTest.java:44]: Resolved

**Comment by KeenanMcCarthy** [comment]
#build
#libby

**Review by tjwatson — APPROVED** [review]

**Pull request merged**